### ISSUE TITLE
[HOLD for payment 2024-06-13] [$250] [Group Chats] [Polish] Move "Leave" button into a row of the Report Details page

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#40256]
As discussed with @shawnborton and @JmillsExpensify we want to update the design of the Report Details page with the following changes:

- Move the "Share" row into the [`ChatDetailsQuickActionsBar`](https://github.com/Expensify/App/blob/a7820d2027138a1a9670b204f608d6b92ec0a454/src/components/ChatDetailsQuickActionsBar.tsx#L15-L16)
- Move the "Leave" button out of this component and into it's own row [here](https://github.com/Expensify/App/blob/a7820d2027138a1a9670b204f608d6b92ec0a454/src/pages/ReportDetailsPage.tsx#L113-L114).

**Result:**

![Image](https://github.com/Expensify/App/assets/32969087/e6c5462b-26bc-406c-a4ba-e25fbf5ba6d5)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b00ddeb3014a078a</li>
        <li>Upwork Job ID: 1779985331229118464</li>
        <li>Last Price Increase: 2024-04-22</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Contributor | 0</li>
<li>ZhenjaHorbach | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @JmillsExpensify</details>

**Comment by JmillsExpensify** [comment]
Going to assign myself as part of the [details revamp](https://docs.google.com/document/d/1-xSZdSZZIuIl0fPO7yypkKfcbWEsnNFAs50ptgL3Gnw/edit) design doc.

**Comment by DylanDylann** [comment]
@marcaaron @JmillsExpensify 
https://github.com/Expensify/App/blob/c75c85f69d46a50258b0cb2870b7b5a4d04702cc/src/pages/ReportDetailsPage.tsx#L323
we only display `ChatDetailsQuickActionsBar` in the report detail If the report is a group chat. Could you help to confirm whether we should display the `ChatDetailsQuickActionsBar` for the room report as well?

**Comment by marcaaron** [comment]
@DylanDylann 

>whether the avatar should be displayed larger

I think we can ignore the avatar for the purposes of this issue. We're focusing on the "Leave" + "Share" actions in this ticket.

>Do we also display the ChatDetailsQuickActionsBar for the room report

We did this largely to focus the introduction of this behavior and allow testing it for Group Chats in isolation. But I think ideally we would take this opportunity to standardize the UI for this. What do you think @JmillsExpensify?

Since the "Leave" action is the only one with a behavior specific to Group Chats - we should be able to show this for Rooms as well. But we can investigate as part of this issue to make sure that assumption is correct.

**Comment by dannymcclain** [comment]
@dragnoir same as the other issue, but posting here for posterity:

![image](https://github.com/Expensify/App/assets/7516624/e8ce31b1-573e-4da5-9c22-70a7e429bcdd)

[Figma link](https://www.figma.com/file/NDPg7ag0bcWW7QjBeSphmf/vip-split%3A-group-chats?type=design&node-id=4037-10516&mode=design&t=5Ofea5c6tqZzk3uz-11)

**Comment by DylanDylann** [comment]
@dannymcclain It means that we still display `ChatDetailsQuickActionsBar` on both group and room detail pages. 
- In the group detail page we will display pin and share option 
- In the room detail page we will display leave and pin option 

cc @marcaaron @JmillsExpensify

**Comment by DylanDylann** [comment]
@dragnoir we should confirm requirements clearly before updating the proposal to save effort 😄

**Comment by dannymcclain** [comment]
Yeah I was gonna say we may wanna confirm with @JmillsExpensify that those are the correct options/buttons before proceeding.

**Comment by marcaaron** [comment]
@DylanDylann re-assigned you if you can push this one forward.

**Comment by DylanDylann** [comment]
@marcaaron Waiting for confirmation before processing as mentioned [here](https://github.com/Expensify/App/issues/40256#issuecomment-2065313876)

**Comment by DylanDylann** [comment]
@dannymcclain @JmillsExpensify any update here?

**Comment by marcaaron** [comment]
My two cents: Doesn't seem like we would want that to be inconsistent between Rooms and Groups.

**Comment by dannymcclain** [comment]
Sorry—missed this one. I think depending on different situations, different buttons show up in that little horizontal group of buttons. There's [a Figma file](https://www.figma.com/file/qYh3d5iJSN2yppP4JsYogt/Details-Clean-up-2024?type=design&node-id=15-7918&mode=design&t=LJDyJ253Af2rJBiw-11) that @shawnborton and @JmillsExpensify have been working on to accompany a bunch of these "detail screen" updates.

@JmillsExpensify mind weighing in here?

**Comment by shawnborton** [comment]
> In the room detail page we will display leave and pin option

This is probably just a mistake from our Figma files. We want leave to always be a row at the bottom of the option rows, it would never be a button in the horizontal button stack.

**Comment by shawnborton** [comment]
Here is a better mockup to reference:
![CleanShot 2024-04-24 at 08 00 05@2x](https://github.com/Expensify/App/assets/2319350/4e286af0-b762-4d99-9800-ae27f78083b0)

**Comment by dannymcclain** [comment]
Thank you Shawn!

**Comment by DylanDylann** [comment]
Thanks for your confirmation. In the scope of this issue, we need to move the "Share" button to `ChatDetailsQuickActionsBar` and move the "Leave" button into a row, also apply for the Room Detail page. @ZhenjaHorbach's [proposal](https://github.com/Expensify/App/issues/40256#issuecomment-2057868506) is enough for this and their proposal is first, let's go with them

**Comment by DylanDylann** [comment]
Thanks for your confirmation. In the scope of this issue, we need to move the "Share" button to `ChatDetailsQuickActionsBar` and move the "Leave" button into a row, also apply for the Room Detail page. @ZhenjaHorbach's [proposal](https://github.com/Expensify/App/issues/40256#issuecomment-2057868506) is enough for this and their proposal is first, let's go with them

🎀👀🎀 C+ reviewed

**Comment by DylanDylann** [comment]
@dragnoir I understand your viewpoint, but this issue doesn't focus on updating to the new UI (you can check the description and title of this issue again). Anyway, this is my decision, let's wait for @marcaaron to give the official assignment

**Comment by DylanDylann** [comment]
Please

> Anyway, this is my decision, let's wait for @marcaaron to give the official assignment

**Comment by marcaaron** [comment]
I think this conversation could be taking a bit of an unproductive turn. I have the following thoughts:

- We will assign @ZhenjaHorbach here. It's the C+ who should recommend someone. I don't feel the need to go against that decision. C+ need to be able to operate with some autonomy here and should be following our rules.
- I don't think this ticket is particularly complex and don't quite feel like anyone is more or less qualified to do it (at least not based on what I can see in this issue). My priority is always to have people work together to implement the best solution.
- @dragnoir an option for you could be to prepare some kind of claim that says "I materially contributed x, y, z to this issue" and reach out to the contributor email. Compensation requests for contributors who go above and beyond are pretty typical around here. We have a history of retaining and rewarding people who help us out. We also have some pretty clear language in the contributing guidelines about all of this IIRC so please review them again at your leisure.

**Comment by JmillsExpensify** [comment]
Still working through PR

**Comment by shawnborton** [comment]
Hmm I don't think I am following, can you rephrase your question? Perhaps @marcaaron knows the answer though.

**Comment by marcaaron** [comment]
Yep, so if we're removing the the "Share" code rows from views that have it those will need to have the "Pin" and "Share" buttons added otherwise there is no way to "share". @DylanDylann is asking if we want to only show the button UI for Group Chats and I believe the answer is "No". Let's make sure it shows up on pretty much all the chats where we currently show a "Share" code.

**Comment by shawnborton** [comment]
Ah, right. We do want to update that everywhere. But I guess I'd be curious to know from @JmillsExpensify - how are we managing these group-specific GHs (that have changes we want everywhere) with your Details clean up project?

**Comment by DylanDylann** [comment]
@shawnborton As I understand, in this issue we will display the share button on the `ChatDetailsQuickActionsBar` in all chats. Do you know how about the pin/unpin button? Currently, we only display the pin/unpin button on the `ChatDetailsQuickActionsBar` if it is a group chat. 

<img width="382" alt="Screenshot 2024-05-09 at 16 32 12" src="https://github.com/Expensify/App/assets/141406735/1eff03eb-dea1-4ae3-8da9-76d625b9631b">


cc @marcaaron

**Comment by shawnborton** [comment]
The pin button will be similar in that we want to update it's location on all things that have the ability to be pinned. So on rooms/reports/expenses/etc, the pin button would move to the promoted actions area. At least that is my understanding!

**Comment by DylanDylann** [comment]
@shawnborton For more understadnable, 

### This is staging

<img width="381" alt="Screenshot 2024-05-10 at 11 30 24" src="https://github.com/Expensify/App/assets/141406735/8f793a44-e584-4aeb-a56a-9ea9bf343111">


### Expect 

<img width="378" alt="Screenshot 2024-05-10 at 11 29 40" src="https://github.com/Expensify/App/assets/141406735/f1d427f9-dda9-42ef-acac-9a062369cd33">

cc @marcaaron

**Comment by shawnborton** [comment]
I guess you can't Join groups or #announce rooms as a heads up.

But @DylanDylann I agree with your "Expect" screenshot, though I think when we have just one button, we use a max-width of 50% of the space or something like that. 

So cc @grgia - just want to make sure you have visibility into what's happening on this issue.

**Comment by shawnborton** [comment]
Okay cool, that makes sense to me.

**Comment by JmillsExpensify** [comment]
> Ah, right. We do want to update that everywhere. But I guess I'd be curious to know from [@JmillsExpensify](https://github.com/JmillsExpensify) - how are we managing these group-specific GHs (that have changes we want everywhere) with your Details clean up project?

@shawnborton good question. This one wasn't on my radar, though what we're doing here will definitely affect what we doing across the platform for the "details revamp." @grgia do you have any recommendations for how we handle this? I think what we're doing here is fine, as long as it's future-compatible with the broader changes we are making soon.

**Comment by DylanDylann** [comment]
@Expensify/design @grgia  Summary: We always display Pin and Share button in the promoted area in every detail report like this 

<img width="801" alt="Screenshot 2024-05-15 at 17 44 05" src="https://github.com/Expensify/App/assets/141406735/8caa03ca-2fe2-4214-8ec2-805ec5b8926a">

Please give a thump up if you agree

cc @ZhenjaHorbach

**Comment by shawnborton** [comment]
We should never have "Leave" as a button - if we do have the option to leave a room, it should show as a row at the bottom.

We've got some nice mockups in [Figma here](https://www.figma.com/design/qYh3d5iJSN2yppP4JsYogt/Details-Clean-up-2024?node-id=15%3A7918&t=gzurHp2KfEpTeVuE-1) for you to see. Just request view access and we'll let you in.

**Comment by greg-schroeder** [comment]
Hi team - should [this issue](https://github.com/Expensify/App/issues/42204) potentially be solved as part of the work that's being done here?

**Comment by marcaaron** [comment]
Looks kind of unrelated to me tbh. We could combine in into this ticket, but might add more confusion than it avoids.

**Comment by greg-schroeder** [comment]
Okay thanks Marc!

**Comment by JmillsExpensify** [comment]
Working through PR.

**Comment by DylanDylann** [comment]
This issue could be included in https://github.com/Expensify/App/issues/42071, I don't think we need to create a regression test for this issue

**Comment by JmillsExpensify** [comment]
Yes, and in fact, we have a whole set of regression tests created for a related project, so I agree that we're covered on that front.

**Comment by JmillsExpensify** [comment]
Payment summary:

 - Contributor: $250 @ZhenjaHorbach 
 - Contributor+:$250 @DylanDylann

**Comment by JmillsExpensify** [comment]
Both contributors paid via Upwork. Closing.
